walk-in-interview/src/main/java/com/google/job/data/JobPayment.java

```diff
@@ -0,0 +1,99 @@
+package com.google.job.data;
+
+/** Class that represents the payment details of a job. */
+public final class JobPayment {
+    private final int min;
+    private final int max;
+    private final PaymentFrequency paymentFrequency;
+
+    private int hashCode;
+
+    private void validateParameter(float min, float max) throws IllegalArgumentException {
+        if (min < 0) {
+            throw new IllegalArgumentException("\"min\" should not be negative");
+        }
+
+        if (max < min) {
+            throw new IllegalArgumentException("\"max\" should not be less than \"min\"");
+        }
+    }
+
+    // For serialization
+    public JobPayment() {
+        this.min = 0;
```

Comment:
you should do:
```
super(/* min= */0, /* max= */0, PaymentFrequency.HOURLY);
```

Reply:
Fixed with `this(/* min= */0, /* max= */0, PaymentFrequency.HOURLY);`